plugins/src/test/java/org/smartregister/fhir/gateway/plugins/UtilsTest.java

```diff
@@ -179,4 +188,77 @@ public void testReadApplicationConfigBinaryResourceWithEmptyResourceId() {
         Assert.assertNull(result);
     }
+
+    @Test
+    public void testGenerateHashConsistency() throws NoSuchAlgorithmException {
+        String input = "consistentTest";
+        String hash1 = Utils.generateHash(input);
+        String hash2 = Utils.generateHash(input);
+        Assert.assertEquals(hash1, hash2);
+    }
+
+    @Test
+    public void testGenerateHashDifferentInputs() throws NoSuchAlgorithmException {
+        String input1 = "inputOne";
+        String input2 = "inputTwo";
+        String hash1 = Utils.generateHash(input1);
+        String hash2 = Utils.generateHash(input2);
+        Assert.assertNotEquals(hash1, hash2);
+    }
+
+    @Test
+    public void testFetchAllBundlePagesAndInject() {
+        Bundle firstPageBundle = new Bundle();
+        firstPageBundle.setMeta(new Meta().setLastUpdated(new Date()));
+        firstPageBundle.addLink().setRelation(Bundle.LINK_NEXT).setUrl("nextPageUrl");
+
+        Bundle secondPageBundle = new Bundle();
+        secondPageBundle.setMeta(new Meta().setLastUpdated(new Date()));
+        secondPageBundle.addEntry(new Bundle.BundleEntryComponent());
+
+        IGetPage loadPageMock = Mockito.mock(IGetPage.class);
+        IGetPageTyped iGetPageTypedMock = Mockito.mock(IGetPageTyped.class);
+        Mockito.doReturn(loadPageMock).when(genericClientMock).loadPage();
+        Mockito.doReturn(iGetPageTypedMock).when(loadPageMock).next(firstPageBundle);
+        Mockito.doReturn(secondPageBundle).when(iGetPageTypedMock).execute();
+        Utils.fetchAllBundlePagesAndInject(genericClientMock, firstPageBundle);
+
+        Assert.assertEquals(1, firstPageBundle.getEntry().size());
+        Assert.assertNull(firstPageBundle.getLink(Bundle.LINK_NEXT));
+        Assert.assertNotNull(firstPageBundle.getMeta().getLastUpdated());
+        Mockito.verify(genericClientMock.loadPage(), Mockito.times(1)).next(firstPageBundle);
+    }
+
+    @Test
+    public void testCleanUpServerBaseUrl_MultipleLinks() {
+        Bundle resultBundle = new Bundle();
+        resultBundle
+                .addLink()
+                .setRelation(Bundle.LINK_NEXT)
+                .setUrl("http://old-base-url/nextPage?param=value");
+        resultBundle.addLink().setRelation(Bundle.LINK_PREV).setUrl("http://old-base-url/prevPage");
+
+        Mockito.when(genericClientMock.getUrlBase()).thenReturn("http://new-base-url");
+
+        Utils.cleanUpServerBaseUrl(genericClientMock, resultBundle);
+
+        List<Bundle.BundleLinkComponent> links = resultBundle.getLink();
+
+        Bundle.BundleLinkComponent nextLink =
+                links.stream()
+                        .filter(link -> Bundle.LINK_NEXT.equals(link.getRelation()))
+                        .findFirst()
+                        .orElse(null);
+        Assert.assertNotNull(nextLink);
+        // TODO the assertion below should pass
+        // Assert.assertEquals("http://new-base-url/nextPage?param=value", nextLink.getUrl());
```

Comment:
The `cleanBaseUrl` function does not consider the path elements when replacing the base url. Is this an expected behavior or should it add them too. For example: `http://old-base-url/nextPage?param=value` should become `http://new-base-url/nextPage?param=value` instead, it becomes `http://new-base-url?param=value`

Reply:
Good catch , though as mentioned the updates for that I will be making on the original PR here - https://github.com/onaio/fhir-gateway-extension/pull/92

Reply:
Also as recommended you can make the PR to point to main instead